### PR TITLE
docs: add DHT groupchats packet protocols to spec

### PR DESCRIPTION
**Pull request by JFreegman**
This change is [Reviewable](https://reviewable.io/reviews/toktok/hs-toxcore/195)

### CODE REVIEW

File: src/Network/Tox/Application/GroupChatsPackets.lhs [hunk 1]
```diff
@@ -0,0 +1,654 @@
+\chapter{DHT Group Chats Packet Protocols}
+
+All packet fields are considred mandatory unless flagged as \textbf{\verb'[optional]'}.
```
Comment (iphydf) [inline src/Network/Tox/Application/GroupChatsPackets.lhs:3]: considered

File: src/Network/Tox/Application/GroupChatsPackets.lhs [hunk 2]
```diff
@@ -0,0 +1,654 @@
...
+\end{tabular}
+
+The \textbf{\verb'Chat ID Hash'} is a
+\href{https://en.wikipedia.org/wiki/Jenkins_hash_function#one_at_a_time}{Jenkins One At A Time Hash}
+of the chat ID. This is used to identify which group a particular message is intended
+for. The \textbf{\verb'sender public encryption key'} is used to identify the peer
+who sent the packet, and the \textbf{\verb'nonce'} is used for decryption.
+
+The encrypted header for lossless and lossy packets contains between 0
+and 8 bytes of empty padding, which is used to mitigate certain types of
```
Comment (iphydf) [inline src/Network/Tox/Application/GroupChatsPackets.lhs:55]: Explain how the number between 0 and 8 is computed. Also specify which crypto attacks it mitigates.

File: src/Network/Tox/Application/GroupChatsPackets.lhs [hunk 3]
```diff
@@ -0,0 +1,654 @@
...
+  \texttt{1}            & \textbf{\verb'is_response'} \\
+  \texttt{32}           & Public Encryption Key \\
+\end{tabular}
+
+Key rotation packets are used to rotate session encryption keys with
+a peer. If \textbf{\verb'is_response'} is false, the packet initiates a
+public key exchange. Otherwise the packet is a response to a previously
+initiated exchange.
+
+The public encryption key must be a brand new, unused key, which takes
```
Comment (iphydf) [inline src/Network/Tox/Application/GroupChatsPackets.lhs:216]: "brand new, unused" could also be "new" or "newly generated", no? "brand new" sounds dramatic :P

**Review by iphydf — CHANGES_REQUESTED** [review]

File: src/Network/Tox/Application/GroupChatsPackets.lhs (hunk 2 above)
Comment (JFreegman) [reply src/Network/Tox/Application/GroupChatsPackets.lhs]: This was copied from https://github.com/TokTok/c-toxcore/blob/master/toxcore/net_crypto.c#L1100

I can't remember the purpose of the padding, and it's not documented elsewhere in the spec.

**Review by iphydf — APPROVED** [review]

**Pull request merged**